test/e2e/azure_test.go

```diff
@@ -112,6 +120,251 @@ var _ = Describe("Workoad cluster creation", func() {
+						MatchLabels: map[string]string{
+							"app": "ingress-nginx",
+						},
+					},
+					Template: corev1.PodTemplateSpec{
+						ObjectMeta: metav1.ObjectMeta{
+							Labels: map[string]string{
+								"app": "ingress-nginx",
+							},
+						},
+						Spec: corev1.PodSpec{
+							Containers: []corev1.Container{
+								{
+									Name:  "web",
+									Image: "nginx:1.18",
+									Ports: []corev1.ContainerPort{
+										{
+											Name:          "http",
+											Protocol:      corev1.ProtocolTCP,
+											ContainerPort: 80,
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			}
+			_, err := deploymentsClient.Create(deployment)
+			Expect(err).NotTo(HaveOccurred())
+			input := framework.WaitForDeploymentsAvailableInput{
+				Getter:     deploymentsClientAdapter{client: deploymentsClient},
+				Deployment: deployment,
+			}
+			framework.WaitForDeploymentsAvailable(context.TODO(), input, e2eConfig.GetIntervals(specName, "wait-deployment")...)
+
+			By("creating an internal Load Balancer service")
+			servicesClient := clientset.CoreV1().Services(corev1.NamespaceDefault)
+			ilbService := &corev1.Service{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "ingress-nginx-ilb",
+					Namespace: corev1.NamespaceDefault,
+					Annotations: map[string]string{
+						"service.beta.kubernetes.io/azure-load-balancer-internal": "true",
+					},
+				},
+				Spec: corev1.ServiceSpec{
+					Type: corev1.ServiceTypeLoadBalancer,
+					Ports: []corev1.ServicePort{
+						{
+							Name:     "http",
+							Port:     80,
+							Protocol: corev1.ProtocolTCP,
+						},
+						{
+							Name:     "https",
+							Port:     443,
+							Protocol: corev1.ProtocolTCP,
+						},
+					},
+					Selector: map[string]string{
+						"app": "ingress-nginx",
+					},
+				},
+			}
+			_, err = servicesClient.Create(ilbService)
+			Expect(err).NotTo(HaveOccurred())
+			ilbSvcInput := WaitForServiceAvailableInput{
+				Getter:  servicesClientAdapter{client: servicesClient},
+				Service: ilbService,
+			}
+			WaitForServiceAvailable(context.TODO(), ilbSvcInput, e2eConfig.GetIntervals(specName, "wait-service")...)
+
+			By("connecting to the internal LB service from a curl pod")
+			jobsClient := clientset.BatchV1().Jobs(corev1.NamespaceDefault)
+			svc, err := servicesClient.Get("ingress-nginx-ilb", metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			var ilbIP string
+			for _, i := range svc.Status.LoadBalancer.Ingress {
+				if net.ParseIP(i.IP) != nil {
+					ilbIP = i.IP
+					break
+				}
+			}
+			ilbJob := &batchv1.Job{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "curl-to-ilb-job",
+					Namespace: corev1.NamespaceDefault,
+				},
+				Spec: batchv1.JobSpec{
+					Template: corev1.PodTemplateSpec{
+						Spec: corev1.PodSpec{
+							RestartPolicy: corev1.RestartPolicyNever,
+							Containers: []corev1.Container{
+								{
+									Name:  "curl",
+									Image: "byrnedo/alpine-curl",
```

Comment:
Are we okay to use random images from dockerhub?

Reply:
I copied this randomness directly from the AKS Engine test, but I'll try to find a more official container image to use.

Reply:
`curlimages/curl` is the official Alpine-based image produced by the curl dev team. I looked for an existing CAPI or Kubernetes test image, and for something usable at mcr.microsoft.com, but didn't find anything. If there's an image we already use in these kind of test scenarios, please clue me in because that would be preferable.